packages/platform-sdk-profiles/src/drivers/memory/contacts/contact-address.ts

```diff
@@ -18,6 +18,9 @@ export class ContactAddress implements IContactAddress {
 	}
 
 	public static async make(data: IContactAddressData): Promise<ContactAddress> {
+		// @TODO: we need access to the profile to create a coin instance here but
```

Comment:
@marianogoldman any idea for a refactor here to make sure we have an instance passed in but not requiring extra work from the client?

Reply:
From the usages (current ones) it seems this is only a helper method you extracted to isolate `ContactAddress` creation, so it won't bother the client, as all they'd do is `contacts().create(...)` on the `ContactAddressRepository`. The repository would be the one to have the coin instance, right? I mean... inside a profile, do we have one contact repo per wallet? one for the whole profile? Or one per coin? If the 1st, we're good, the repo should have the instance and pass it on. If the last one, probably the same. Problem would be if it was a singleton for the profile (which I think it's not).

Reply:
Contacts are per-profile data, they are not attached to any specific wallet.